packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.spec.ts

```diff
@@ -27,6 +27,18 @@ describe('getActionNameFromElement', () => {
     expect(getActionNameFromElement(element`<div>Foo <div>bar</div></div>`)).toBe('Foo bar')
   })
 
+  it('replaces the programmatic action name in textual content', () => {
+    expect(getActionNameFromElement(element`<div>Foo <div data-dd-action-name="custom action">bar<div></div>`)).toBe(
+      'Foo custom action'
+    )
+  })
+
+  it('replaces the programmatic action name in textual content based on the user-defined attribute', () => {
+    expect(
+      getActionNameFromElement(element`<div>Foo <div data-test-id="custom action">bar<div></div>`, 'data-test-id')
+    ).toBe('Foo custom action')
+  })
+
```

Comment:
nit: for consistency with the file organization, those tests should be in the `programmatically declared action name` suite